File: 2020/00/main.java
import tcollier.Executor;
import tcollier.Solution;

class Day0Solution implements Solution {
  private String[] data;

  public Day0Solution(String[] data) {
    this.data = data;
  }

  public String part1Answer() {
    return this.data[0];
  }

  public String part2Answer() {
    return this.data[1];
  }
}

class Main {
  public static void main(String[] args) {
    String[] data = {"Hello", "World!"};
    Executor executor = new Executor(new Day0Solution(data));
    executor.run(args);
  }
}

File: 2020/00/main.scala
import tcollier.Executor
import tcollier.Solution

class Day0Solution extends Solution[String] {
  def part1Answer(input: Array[String]): String = {
    return input(0)
  }

  def part2Answer(input: Array[String]): String = {
    return input(1)
  }
}

object Main {
  def main(args: Array[String]) = {
    val input: Array[String] = Array("Hello", "World!")
    val executor = new Executor[String](new Day0Solution(), input)
    executor.run(args)
  }
}
